src/components/Personalization/createRedirectHandler.js

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2021 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+const getRedirectDetails = redirectDecisions => {
+  const decision = redirectDecisions[0];
+  const { items, id, scope } = decision;
+  const { content } = items[0].data;
+
+  return { content, decisions: [{ id, scope }] };
+};
+
+export default ({ collect, win = window }) => {
+  return redirectDecisions => {
+    const { content, decisions } = getRedirectDetails(redirectDecisions);
+    const documentMayUnload = true;
+
+    return collect({ meta: { decisions } }, documentMayUnload).then(() => {
+      win.location.replace(content);
```

Comment:
I'm noticing this in the [docs for `location.replace`](https://developer.mozilla.org/en-US/docs/Web/API/Location/replace):

> If the assignment can't happen because of a security violation, a DOMException of the SECURITY_ERROR type is thrown. This happens if the origin of the script calling the method is different from the origin of the page originally described by the Location object, mostly when the script is hosted on a different domain.

I haven't verified, but that seems to suggest that if Alloy is loaded from a different origin than the page loading Alloy, that an error will be thrown when Alloy calls `location.replace`. At least, that's my interpretation. Have you tested this scenario?

Reply:
a try/catch wouldn't cause any harm, but currently, we don't try-catch in at.js. I will add a try/catch, it would catch exceptions like malformed URL as well.

Reply:
I'm more concerned that the navigation won't actually occur. You've verified that the navigation actually occurs in this scenario though? I'm probably just misunderstanding MDN's documentation, but now I want to know what they're documentation is trying to say. If no error is being thrown in this scenario and it's actually navigating the user, then there's no need to add a catch block unless we know what/why we're catching.

Reply:
Ignore my last sentence. It might still valuable to catch. We can leave it. Is it a problem that if we try to redirect and it failes, that we show container but won't have stored any of the views in the cache?